### PR TITLE
cel: patch thread safety issue

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Commit Message: Disable fast-path in CEL protobuf wrapper. The fast path uses internal protobuf methods via a private friend. These methods are not thread-safe, and cause an assertion failure when applied to protobufs shared across workers, such as cluster metadata. The fast path will be enabled again once both protobuf and CEL are updated.
Risk Level: low, switches to a different implementation
Testing: none added
Docs Changes: none
Release Notes: none
Reference: https://github.com/istio/istio/issues/28175

### CODE REVIEW

**Comment by moderation** [comment]
If @lizan can resolve the test issues with https://github.com/envoyproxy/envoy/pull/13712, is this PR necessary? Would be good to minimize patches if possible

**Comment by kyessenov** [comment]
@moderation yes. Internal fix is not ready.

**Comment by PiotrSikora** [comment]
@kyessenov RBE cache is busted, you need to merge `master` to re-kick the macOS test.

**Comment by lizan** [comment]
@moderation or @htuch are you ok with this?

**Review by moderation — CHANGES_REQUESTED** [review]
A TODO explaining this patch is required until the new protobuf is released would be good

File: bazel/repositories.bzl [hunk 1]
```diff
@@ -348,7 +348,12 @@ def _com_github_zlib_ng_zlib_ng():
     )
 
 def _com_google_cel_cpp():
-    external_http_archive("com_google_cel_cpp")
+    external_http_archive(
+        "com_google_cel_cpp",
+        patch_args = ["-p1"],
+        # Patches to remove "fast" protobuf-internal access
+        patches = ["@envoy//bazel:cel-cpp.patch"],
+    )
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply bazel/repositories.bzl]: Added an explanation.

**Review by lizan — APPROVED** [review]

**Pull request merged**